lib/hexapdf/type/image.rb

```diff
@@ -254,6 +257,48 @@ def png_chunk(type, data = '')
         [data.length].pack("N") << type << data << [Zlib.crc32(data, Zlib.crc32(type))].pack("N")
       end
 
+      # Writes the image as TIFF to the given IO stream.
+      def write_tiff(io, info)
+        stream = stream_source
+
+        # header
+        io << ['49492A00'].pack('H*')
+
+        # offset to IFD
+        pad_size = stream.length % 2
+        io << [stream.length + 8 + pad_size].pack('L')
+
+        # data
+        io << Filter.string_from_source(stream)
+
+        # pad to word boundary
+        io << [0].pack('C') * pad_size
+
+        decode_params = [self[:DecodeParms]].flatten[0]
+        compression = if decode_params[:K] > 0 # G3-2D
+                        3
+                      elsif decode_params[:K] < 0 # G4-2D
+                        4
+                      else # G3-1D
+                        2
+                      end
+
+        # IFD entries
+        ifd_entries = [
+            [0x0100, 3, 1, info.width], # width
+            [0x0101, 3, 1, info.height], # height
```

Comment:
Shouldn't we better use LONG values instead of SHORT for width and height? Just to be safe in case of really big images?